common/models/create_environment_request.go

```diff
@@ -21,16 +22,13 @@ func (r CreateEnvironmentRequest) Validate() error {
 		return fmt.Errorf("EnvironmentName is required")
 	}
 
-	if r.MinScale < 0 {
-		return fmt.Errorf("MinScale must be a positive integer")
+	if r.Scale < 0 {
+		return fmt.Errorf("Scale must be a positive integer")
 	}
 
-	if r.MaxScale < 0 {
-		return fmt.Errorf("MaxScale must be a positive integer")
-	}
-
-	if r.MaxScale < r.MinScale {
-		return fmt.Errorf("MaxScale must be greater than or equal to MinScale")
+	if strings.EqualFold(r.EnvironmentType, EnvironmentTypeDynamic) &&
+		!strings.EqualFold(r.OperatingSystem, LinuxOS) {
+		return fmt.Errorf("%s is not a supported OS for dynamic environments", r.OperatingSystem)
```

Comment:
Seems like we are missing some other error scenarios:
* `EnvironmentType == EnvironmentTypeDynamic && Scale != 0`
* `!(EnvironmentType == EnvironmentTypeDynamic || EnvironmentType == EnvironmentTypeStatic)`

Reply:
I realize that since `EnvironmentType` has a default, our pattern has been to allow empty strings in that field, so while it may make sense to not do that validation here, it still needs to be done somewhere. 
Also, we may want to reconsider allowing empty strings like that (the main reason I used that pattern is because I was impatient adding each field in swagger; definitely not the best reason to keep it around).

Reply:
If the `EnvironmentType` is explicitly set, I'm ignoring `Scale` intentionally. But I can see the value in informing the user when this is the case. ✅ 
> EnvironmentType == EnvironmentTypeDynamic && Scale != 0 

If we do, then it would make sense to call Validate from the Terraform provider to right?

Reply:
I think adding that validation in, and including it in terraform, would make our code better.

Reply:
✅